refactoring of battleship_oo2

pythonlectures opened the pull request:
added comments
some variables are kept private via _ convention
added Cell class

maveec committed:
refactoring of battleship_oo2
added comments
some variables are kept private via _ convention
added Cell class

maveec committed:
refactoring of battleship_oo2
added comments
some variables are kept private via _ convention
added Cell class

maveec committed:
test to check file ending in github

maveec committed:
refactoring of battleship_oo2_test.py
added setters and getters

maveec committed:
bug fixing

maveec committed:
message taken out to Game

maveec committed:
made everything private

mocza commented on battleship_oo2_test.py:
```diff
@@ -11,8 +11,8 @@ def __init__(self, player1: 'Player', player2: 'Player'):
+        self._shooter = self._player1
+        self._receiver = self._player2
 
     def play_game(self):
```
This is the only method I can see you use from outside Game which means all other Game methods should be private.
I think there are other examples for the same ie. public methods that should be private.

Consider creating all methods as private by default and only make those methods public that you especially want to expose to the outside world.

mocza commented on battleship_oo2_test.py:
```diff
@@ -118,17 +115,40 @@ def take_shot(self, shot):
+    def __init__(self, coordinates: 'Coordinates', status: str = None):
+        self._coordinates = coordinates
+        self._status = status
+        self._allowed_statuses = {'Hit', 'Not Hit'}
```
Consider using an enum for constants like 'Hit' and 'Not Hit' to eliminate hardcoding these strings multiple times in the code. With an enum you can define valid statuses and use those same enum values across the board.

mocza commented on battleship_oo2_test.py:
```diff
@@ -87,29 +78,35 @@ class Ship:
-    def __init__(self, board_len):
+
+    def __init__(self, board_len: int):
         self.coordinates = [Coordinates(x=x, y=y) for x in range(1, board_len) for y in range(1, board_len)]
```
You populate coordinates but don't use it anywhere instead you use the shots_taken set.
Although this works with the current features, you should remove any unused code and I suggest you remove the shots_taken set and keep coordinates storing Cells instead of Coordinates. 

If the Board consists of Cells you can register a shot on the board by setting the status of the cell to something like 'Undiscovered' by default and make it 'Empty' or 'Hit' after a shot and it's also easy to verify if a shot is valid on the board or not.

maveec committed:
made methods private

maveec committed:
Use of Enum for cellStatus

maveec committed:
add_shot removed
Board now contains Cells instead of Coordinates

maveec committed:
code formatting

maveec committed:
lined battleship_precedural2 to battleship_oop2_test

maveec committed:
CRLF for battleship_oo2

pythonlectures committed:
cleaned 002 test

pythonlectures committed:
added error handling for input

pythonlectures committed:
variant for tests. still not sure when to use fixtures and when to aggregate asserts in one function

pythonlectures committed:
added README.md file

pythonlectures committed:
packaged tests folder
cfr https://stackoverflow.com/questions/41748464/pytest-cannot-import-module-while-python-can
